internal/cli/ops_manager_clusters_indexes_create.go

```diff
@@ -0,0 +1,189 @@
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) init() error {
+	if opts.ProjectID() == "" {
+		return errMissingProjectID
+	}
+
+	var err error
+	opts.store, err = store.New()
+	return err
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) Run() error {
+	current, err := opts.store.GetAutomationConfig(opts.ProjectID())
+	if err != nil {
+		return err
+	}
+
+	index, err := opts.newIndex()
+	if err != nil {
+		return err
+	}
+
+	current.IndexConfigs = index
+
+	if err = opts.store.UpdateAutomationConfig(opts.ProjectID(), current); err != nil {
+		return err
+	}
+
+	fmt.Print(messages.DeploymentStatus(config.OpsManagerURL(), opts.ProjectID()))
+
+	return nil
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) newIndex() ([]*om.IndexConfigs, error) {
+	keys, err := opts.indexKeys()
+	if err != nil {
+		return nil, err
+	}
+
+	i := new(om.IndexConfigs)
+	i.DBName = opts.db
+	i.CollectionName = opts.collection
+	i.RSName = opts.rsName
+	i.Key = keys
+	i.Options = opts.newIndexOptions()
+
+	if opts.locale != "" {
+		i.Collation = opts.newCollationOptions()
+	}
+
+	return []*om.IndexConfigs{i}, nil
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) newIndexOptions() *atlas.IndexOptions {
+	return &atlas.IndexOptions{
+		Background: opts.background,
+		Unique:     opts.unique,
+		Sparse:     opts.sparse,
+		Name:       opts.name,
+	}
+}
+
+func (opts *opsManagerClustersIndexesCreateOpts) newCollationOptions() *atlas.CollationOptions {
+	return &atlas.CollationOptions{
+		Locale:          opts.locale,
+		CaseLevel:       opts.caseLevel,
+		CaseFirst:       opts.caseFirst,
+		Strength:        opts.strength,
+		NumericOrdering: opts.numericOrdering,
+		Alternate:       opts.alternate,
+		MaxVariable:     opts.maxVariable,
+		Normalization:   opts.normalization,
+		Backwards:       opts.backwards,
+	}
+}
+
+// indexKeys takes a slice of values formatted as key:vale and returns an array of slice [[key, value][key, value]]
+func (opts *opsManagerClustersIndexesCreateOpts) indexKeys() ([][]string, error) {
+	propertiesList := make([][]string, len(opts.keys))
+	for i, key := range opts.keys {
+		value := strings.Split(key, ":")
+		if len(value) != 2 {
+			return nil, fmt.Errorf("unexpected key format: %s", key)
+		}
+		values := []string{value[0], value[1]}
+		propertiesList[i] = values
+	}
+
+	return propertiesList, nil
+}
+
+// mongocli cloud-manager cluster(s) index(es) create [name]  --rsName rsName --dbName dbName [--key field:type] --projectId projectId
+// --locale locale --caseFirst caseFirst --alternate alternate --maxVariable maxVariable --strength strength --caseLevel caseLevel --numericOrdering numericOrdering
+// --normalization normalization --backwards backwards --unique unique --sparse sparse --background background
+func OpsManagerClustersIndexesCreateBuilder() *cobra.Command {
+	opts := &opsManagerClustersIndexesCreateOpts{}
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: description.CreateCluster,
```

Comment:
copy/paste?

Reply:
I got that description from `AtlasClustersIndexesCreateBuilder` 🤔

Reply:
then we both have a copy paste error :P